tabbycat/adjfeedback/management/commands/generatefeedback.py

```diff
@@ -88,24 +88,19 @@ def handle(self, *args, **options):
                                            "skipping".format(debate_id, tournament.slug)))
                 self.handle_debate(debate, **options)
 
-    def handle_round(self, round, **options):
+    def handle_object(self, obj, name_field, delete_func, add_func, **options):
         if options["clean"]:
-            self.stdout.write(self.style.WARNING("Deleting all feedback for round {}...".format(round.name)))
-            delete_all_feedback_for_round(round)
+            self.stdout.write(self.style.WARNING("Deleting all feedback for round {}...".format(getattr(obj, name_field))))
+            delete_func(obj)
 
-        self.stdout.write(self.style.MIGRATE_HEADING("Generating feedback for round {}...".format(round.name)))
+        self.stdout.write(self.style.MIGRATE_HEADING("Generating feedback for round {}...".format(getattr(obj, name_field))))
         try:
-            add_feedback_to_round(round, **self.feedback_kwargs(options))
+            add_func(obj, **self.feedback_kwargs(options))
         except ValueError as e:
             raise CommandError(e)
 
-    def handle_debate(self, debate, **options):
-        if options["clean"]:
-            self.stdout.write(self.style.WARNING("Deleting all feedback for debate {}...".format(debate.matchup)))
-            delete_feedback(debate)
+    def handle_round(self, round, **options):
+        return self.handle_object(round, 'name', delete_all_feedback_for_round, add_feedback_to_round, **options)
 
-        self.stdout.write(self.style.MIGRATE_HEADING("Generating feedback for debate {}...".format(debate.matchup)))
-        try:
-            add_feedback(debate, **self.feedback_kwargs(options))
-        except ValueError as e:
-            raise CommandError(e)
+    def handle_debate(self, debate, **options):
+        return self.handle_object(debate, 'matchup', delete_feedback, add_feedback, **options)
```

Comment:
While this would clearly be flagged by Code Climate, I'm in two minds about whether this refactor makes the code easier or harder to follow. It's in that region where it's short enough to understand quickly by reading, and the two instances are next to each other, and there's a (nonzero, finite) cost in readability and maintainability to using tools like `getattr()`, so it's a pretty borderline call. Curious as to what your opinion is?

Thinking out loud: An alternative—slightly more verbose, but arguably clearer—would be to pass in a lambda function `lambda x: x.name` or equivalently `attrgetter('name')` (from the `operator` module). I consider this to be highly debatable, though, as in, potentially worse.

Also, you missed generalizing the word "round" in the printed strings (it's "debate" in the other one).

Reply:
I think it's still well readable by following the calls: This method calls these other standard ones but adds some control. I like this `attrgetter()`! Also won't have to remember to modify similar methods. (Think the draw consumer is a tighter call; wouldn't combine the teams method)

I've also removed the model name from the print as it would be clear from the object name.